app/src/main/java/waiter/Threadable/mock/ThreadGeneratorMock.java

```diff
@@ -0,0 +1,21 @@
+package waiter.Threadable.mock;
+
+import waiter.Threadable.Threadable;
+
+public class ThreadGeneratorMock implements Threadable {
+
+    private final int numberOfThreadsToGenerate;
+    public int numberOfThreadsGenerated;
+
+    public ThreadGeneratorMock(int numberOfThreadsToGenerate) {
+        this.numberOfThreadsToGenerate = numberOfThreadsToGenerate;
+        this.numberOfThreadsGenerated = 0;
+    }
+
+    public void generate(Runnable runnable) {
+        while(this.numberOfThreadsGenerated < this.numberOfThreadsToGenerate) {
```

Comment:
Hmmm, Id say we don't need it per say, it just allows for a little more verbose test coverage as now the mock is configurable with the number of times it can be run and therefore can use property based testing. I can make it just run a hard coded number more than one (2 or 3) number of times instead, but I thought this was a little more verbose when it came to triangulation for the CommunicatorTest. I hope I explained this well, What are your thoughts? 😄

Reply:
The generate method is called in a while loop in `Communicator` so we can just have the injected `ThreadGeneratorMock` increment a variable `numberofThread` and increment that every time `generate` is called?